### PR TITLE
Refactors validators and adds dark validator.

### LINKED ISSUE #377: Move validate dark service to cocina models
So that it is performed on every instantiation rather than just update/create.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #377

**NOTE:  Changes to openapi.yml require updating openapi.yml for sdr-api and dor-services-app and generating models - see README.**

## Why was this change made? 🤔
To move semantic validation as close to data creation as possible.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

Unit

### CODE REVIEW

File: lib/cocina/generator/generator.rb [hunk 1]
```diff
@@ -78,7 +78,6 @@ def generate_for(schema)
         'rights_description_builder.rb',
         'title_builder.rb',
         'validatable.rb',
-        'validator.rb',
```
Comment (mjgiarlo) [inline lib/cocina/generator/generator.rb:81]: Presumably we don't need to add the new validators to this list because they're in their own subdirectory?

Comment (justinlittman) [reply lib/cocina/generator/generator.rb:81]: As it turns out, yes.

File: lib/cocina/models/validatable.rb [hunk 2]
```diff
@@ -8,15 +8,15 @@ module Validatable
...
+          Validators::Validator.validate(self, attributes.with_indifferent_access) if validate
           super(attributes, safe, &block)
         end
       end
 
       def new(*args)
         validate = args.first.delete(:validate) if args.present?
         new_model = super(*args)
-        Validator.validate(new_model.class, new_model.to_h) if (validate || validate.nil?) && self.class.name
+        Validators::Validator.validate(new_model.class, new_model.to_h) if validate || validate.nil?
```
[the remark this answers is not in the document]
Comment (justinlittman) [reply lib/cocina/models/validatable.rb:19]: I was trying to figure out what it did and realized that it did nothing.

**Review by mjgiarlo — APPROVED** [review]

File: lib/cocina/models/validators/dark_validator.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+        def meets_preconditions?
+          dro? && attributes.dig(:access, :view) == 'dark'
+        end
+
+        def dro?
+          (clazz::TYPES & DRO::TYPES).any?
+        rescue NameError
+          false
+        end
```
Comment (mjgiarlo) [inline lib/cocina/models/validators/dark_validator.rb:38]: @justinlittman I'm wondering if we can call `clazz.dro?` and lean on the `Checkable` module for this instead.

Comment (mjgiarlo) [reply lib/cocina/models/validators/dark_validator.rb:38]: Aha, it's because it's an instance method and we have but a class and some props. Nevermind!

**Pull request merged**